salt/haproxy/init.sls

```diff
@@ -0,0 +1,48 @@
+---
+/etc/haproxy/haproxy.cfg:
+  file.managed:
+    - source: salt://haproxy/haproxy.cfg.jinja
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+
+haproxy:
+  file.managed:
+    - name: /etc/kubernetes/manifests/haproxy.yaml
+    - source: salt://haproxy/haproxy.manifest
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+{% if "kube-master" in salt['grains.get']('roles', []) %}
+  iptables.append:
+{% else %}
+  iptables.delete:
+{% endif %}
+    - table:      filter
+    - family:     ipv4
+    - chain:      INPUT
+    - jump:       ACCEPT
+    - match:      state
+    - connstate:  NEW
+    - dports:
+      - {{ pillar['api']['ssl_port'] }}
+    - proto:      tcp
+
+# Send a HUP to haproxy when the config changes
+# TODO: There should be a better way to handle this, but currently, there is not. See
+# kubernetes/kubernetes#24957
+haproxy_restart:
+  cmd.run:
+    - name: |-
+        haproxy_id=$(docker ps -a | grep -E "k8s_haproxy.*\.infra\.caasp\.local_kube-system_" | awk '{print $1}')
```

Comment:
Minor: we only care about running containers so we should use `docker ps` instead of `docker ps -a`.